### ISSUE TITLE
False positive for react/no-multiline-comp

### ISSUE THREAD

**Issue by tnorling** [jsx-eslint/eslint-plugin-react#3308]
After upgrading to version 7.30.0 from 7.29.4 I'm now getting errors for `react/no-multiline-comp` for functions which are not supposed to be React components and which didn't throw previously. Functions which return null and are defined as an object property, seem to be interpreted as React components now, but shouldn't be.

A minimal example demonstrating this behavior:

```javascript
const testObject = {
  testFunction1: () => {
    return null;
  },
  testFunction2: () => {
    return null;
  }
}
```

**Comment by ljharb** [comment]
I believe this will be fixed in the next release, by #3291.

**Comment by tnorling** [comment]
Thanks!

**Comment by tnorling** [comment]
@ljharb This is still a problem in 7.30.1

**Comment by ljharb** [comment]
Hopefully #3347 will fix it in the next release, then :-)
